### PR TITLE
select2_from_array fix when multiple and X clicked twice

### LINKED ISSUE #1224: [3.6][Bug] select2_from_array with allows_multiple "null" saves incorrectly
# Bug report
When using the select2_from_array field type. If you clear all existing (saved) values and update, the database is left with `[null]` rather than a real `NULL` value.

### What I did:
Field configuration:
```
[
      'name' => 'benefit_exclusions',
      'label' => "Exclude Benefits From This Market",
      'type' => 'select2_from_array',
      'options' => $benefitsArray,
      'allows_null' => true,
      'allows_multiple' => true,
],
```
In the model I also have:
```
protected $casts = [
      'benefit_exclusions' => 'array',
];
```

### What I expected to happen:
Save with a `NULL` value.

### What happened:
It saves with a value of `[null]`

### What I've already tried to fix it:
As a temporary workaround, I changed my update function at the bottom of my CrudController to this:
```
public function update(UpdateRequest $request)
{
    $redirect_location = parent::updateCrud($request);
    if (empty($request->benefit_exclusions)) {
        $this->crud->entry->benefit_exclusions = null;
        $this->crud->entry->save();
    }
    return $redirect_location;
}
```

### Backpack, Laravel, PHP, DB version:
backpack/base: 0.7.25
backpack/crud: 3.2
Laravel: 5.5.24
PHP: 7.1.8
MySQL: 5.6.35

### PR DESCRIPTION
**Pull request by tabacitu**
Fixes #1224 by basically doing #3243 all over again. That PR #3243 has been merged by mistake, then reverted, so it couldn't be merged again. This PR, if merged, will do the exact same thing as that one.

Here's a copy-paste from @pxpm 's #3243 :

----

This is a fix for #1224

It's an old issue, and I think it's a problem that persists with all select2 that allows the clear of all options.

The main problem is that a select multiple without selected value does not get posted by the browser.

We need to send an empty value (so laravel will convert empty to null with ConvertEmptyStringsToNull) thus updating the value in the database accordingly.

The easiest solution would be to have an hidden input that gets sent in case nothing is selected in the multiple select, but that does not work for us because of repeatable.

This solution might be applicable to all select2 inputs.

### CODE REVIEW

**Comment by ian-patel** [comment]
Important Issue! Making my life hard.

**Comment by pxpm** [comment]
Hello @ian-patel, hope you are doing fine.

You can easily have this fixed in your project by just adding this file in your `resources/views/vendor/backpack/crud/fields`, you can do `php artisan backpack:publish fields/select2_from_array` to publish the file into that directory and then replace the contents with this PR code.

When we merge this you could just delete the file and backpack will pick the file from package again. 

Thanks,
Pedro

**Comment by tabacitu** [comment]
I think you'll be happy to find out it's finally time to fix this @pxpm 😅 I know it's been your pet peeve for a while now 😅

I'm still wondering _if_ or _how_ we can do the same, but without all this added JS... it seems like so much code for such a minor bug... 

@promatik what do you think, any ideas on an alternative?

**Comment by promatik** [comment]
@tabacitu can we change the merge branch to 4.2? So we don't have 74 files changed with that StyleCI bug? 😅

File: src/resources/views/crud/fields/select2_from_array.blade.php [hunk 1]
```diff
@@ -1,20 +1,27 @@
-<!-- select2 from array -->
 @php
```
Comment (tabacitu) [inline src/resources/views/crud/fields/select2_from_array.blade.php:1]:
```suggestion
<!-- select2 from array -->
@php
```

File: src/resources/views/crud/fields/select2_from_array.blade.php [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+    $field['placeholder'] = $field['placeholder'] ?? '-';
     $field['allows_null'] = $field['allows_null'] ?? $crud->model::isColumnNullable($field['name']);
+    $field['allows_multiple'] = $field['allows_multiple'] ?? false;
+
 @endphp
+<!-- select2 from array -->
```
Comment (tabacitu) [inline src/resources/views/crud/fields/select2_from_array.blade.php:7]:
```suggestion

```

**Comment by tabacitu** [comment]
@promatik just did that, but I still couldn't get rid of the StyleCI changes. So I opened yet another PR (https://github.com/Laravel-Backpack/CRUD/pull/3941) with changes only in `select2_from_array.blade.php`.

Closing - let's move the conversation to https://github.com/Laravel-Backpack/CRUD/pull/3941

**Pull request closed without merging**